api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2774,27 +2901,198 @@ public void putObject(String bucketName, String objectName, InputStream stream,
+   *       encryptionMaterials);
+   *   bais.close();
+   *   System.out.println("my-bucketname is uploaded successfully");
+   * }
+   * </pre>
+   *
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name to create in the bucket.
+   * @param stream
+   *          stream to upload.
+   * @param size
+   *          Size of all the data that will be uploaded.
+   * @param contentType
+   *          Content type of the stream.
+   * @param encryptionMaterials
+   *          Encryption metadata details and operations.
+   *
+   * @throws InvalidBucketNameException
+   *           upon invalid bucket name is given
+   * @throws NoResponseException
+   *           upon no response from server
+   * @throws IOException
+   *           upon connection error
+   * @throws XmlPullParserException
+   *           upon parsing response xml
+   * @throws ErrorResponseException
+   *           upon unsuccessful execution
+   * @throws InternalException
+   *           upon internal library error
+   * @throws InvalidAlgorithmParameterException
+   *           upon wrong encryption algorithm used
+   * @throws BadPaddingException
+   *           upon incorrect padding in a block
+   * @throws IllegalBlockSizeException
+   *           upon incorrect block
+   * @throws NoSuchPaddingException
+   *           upon wrong padding type specified
+   *
+   * @see #putObject(String bucketName, String objectName, String fileName, String contentType)
+   * @see #putObject(String bucketName, String objectName, String fileName)
+   * @see #putObject(String bucketName, String objectName, InputStream stream, long size, String contentType)
+   *
+   */
+  public void putObject(String bucketName, String objectName, InputStream stream, long size, String contentType,
+      EncryptionMaterials encryptionMaterials) throws InvalidBucketNameException, NoSuchAlgorithmException,
+      InsufficientDataException, IOException, InvalidKeyException, NoResponseException, XmlPullParserException,
+      ErrorResponseException, InternalException, InvalidArgumentException, NoSuchPaddingException,
+      IllegalBlockSizeException, BadPaddingException, InvalidAlgorithmParameterException {
+
+    if (encryptionMaterials == null) {
+      throw new InvalidArgumentException("empty encryption properties not allowed");
+    }
+
+    // Generate symmetric 256 bit AES key for data encryption
+    KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+    symKeyGenerator.init(256);
+    SecretKey dataEncryptionKey = symKeyGenerator.generateKey();
+
+    // Generate an iv to be used for data encryption
+    SecureRandom ivSeed = new SecureRandom();
+    byte[] iv = new byte[16];
+    ivSeed.nextBytes(iv);
+
+    // Get CipherStream after encryption of input stream
+    CipherInputStream cipherInputStream = encryptionMaterials.encryptInputStream(stream, dataEncryptionKey, iv);
```

Comment:
Yes, `CipherStream` is a part of this PR only, but the changes you are proposing will be needed in all the `put object versions`. I meant this is not specific to client side encryption.